docs/plugins/repository-gcs.asciidoc

```diff
@@ -116,36 +116,90 @@ PUT _snapshot/my_gcs_repository
 // CONSOLE
 // TEST[skip:we don't have gcs setup while testing this]
 
-[[repository-gcs-bucket-permission]]
-===== Set Bucket Permission
+[[repository-gcs-client]]
+==== Client Settings
 
-The service account used to access the bucket must have the "Writer" access to the bucket:
+The client used to connect to Google Cloud Storage has a number of settings available.
+Client setting names are of the form `gcs.client.CLIENT_NAME.SETTING_NAME` and specified
+inside `elasticsearch.yml`. The default client name looked up by a `gcs` repository is
+called `default`, but can be customized with the repository setting `client`.
 
-1. Connect to the https://console.cloud.google.com/[Google Cloud Platform Console]
-2. Select your project
-3. Got to the https://console.cloud.google.com/storage/browser[Storage Browser]
-4. Select the bucket and "Edit bucket permission"
-5. The service account must be configured as a "User" with "Writer" access
+For example:
+
+[source,js]
+----
+PUT _snapshot/my_gcs_repository
+{
+  "type": "gcs",
+  "settings": {
+    "bucket": "my_bucket",
+    "client": "my_alternate_client"
+  }
+}
+----
+// CONSOLE
+// TEST[skip:we don't have gcs setup while testing this]
+
+Some settings are sensitive and must be stored in the
+{ref}/secure-settings.html[elasticsearch keystore]. This is the case for the service account file:
+
+[source,sh]
+----
+bin/elasticsearch-keystore add gcs.client.default.credentials_file
```

Comment:
The subcommand is `add-file`, not `add`.

Reply:
Good catch